app/main/controller/presto_controller.py

```diff
@@ -0,0 +1,20 @@
+from flask import request, current_app as app
+from flask_restplus import Resource, Namespace
+
+from app.main.lib.fields import JsonObject
+from app.main.lib import similarity
+
+api = Namespace('presto', description='presto callback url')
+presto = api.model('presto', {
+    'body': JsonObject(required=False, description='Original body of the message sent - will contain ID, callback_url, and original url/text of request'),
+    'response': JsonObject(required=False, description='The response data sent back from the model'),
+})
+@api.route('/receive/<string:action>/<string:model_type>')
+class PrestoResource(Resource):
+    @api.response(200, 'Successfully responded to presto callback.')
+    @api.doc('Receive a presto callback for a given `model_type`')
+    def post(self, action, model_type):
+        if action == "add":
+            return similarity.add_item(request.args or request.json, model_type)
```

Comment:
I think this may be replying to Presto and not the original caller we want to reply to (e.g., check API). I suspect we need to have some sort of internal redis queue as we do with the shared model servers to receive the callback and enqueue the result back to the original process that is handling the request from Check API unless we change Check API to a callback architecture as well

Reply:
discussion today was on changing "everything" to callbacks: https://docs.google.com/document/d/1kIZSy2rQ7npCo_u5NPfyoFoEgLdB6vyJ--mnDxn9SDc/edit#heading=h.hndo1trhkuyf